File: library/src/main/java/com/apppoweron/circularrevealbutton/container/CircularRevealContainer.java
package com.apppoweron.circularrevealbutton.container;

import android.os.Build;
import android.support.annotation.RequiresApi;

import com.apppoweron.circularrevealbutton.ButtonAnimationEndListener;

public interface CircularRevealContainer {
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    void startCircularRevealAnimation(ButtonCRevealAnimationData data, ButtonAnimationEndListener animEndListener);
    void removeAnimationMask();
    void getLocationInWindow(int[] outLocation);
}

File: library/src/main/kotlin/com/apppoweron/circularrevealbutton/container/CircularRevealContainer.kt
package com.apppoweron.circularrevealbutton.container

import android.os.Build
import android.support.annotation.RequiresApi

interface CircularRevealContainer {
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    fun startCircularRevealAnimation(data: ButtonCRevealAnimationData, listener: ((viewId: Int) -> Unit)?)

    fun removeAnimationMask()
    fun getLocationInWindow(outLocation: IntArray)
}
